server/vertx-server/src/main/scala/sttp/tapir/server/vertx/streams/package.scala

```diff
@@ -21,7 +25,68 @@ package object streams {
       readStream: ReadStream[WebSocketFrame],
       pipe: streams.Pipe[REQ, RESP],
       o: WebSocketBodyOutput[streams.Pipe[REQ, RESP], REQ, RESP, _, VertxStreams]
-    ): ReadStream[WebSocketFrame] =
-      throw new UnsupportedOperationException()
+    ): ReadStream[WebSocketFrame] = {
+      val stream0 = optionallyContatenateFrames(readStream, o.concatenateFragmentedFrames)
+      val stream1 = optionallyIgnorePong(stream0, o.ignorePong)
+      val stream2 = optionallyAutoPing(stream1, o.autoPing)
+
+      val stream3 = new ReadStreamMapping(stream2, { (frame:WebSocketFrame) =>
+        o.requests.decode(frame) match {
+          case failure: DecodeResult.Failure =>
+            throw (new WebSocketFrameDecodeFailure(frame, failure))
+          case DecodeResult.Value(v) => v
+        }
+      })
+
+      new ReadStreamMapping(pipe(stream3), o.responses.encode)
+    }
+  }
+
+  private def optionallyContatenateFrames(rs: ReadStream[WebSocketFrame], doConcatenate: Boolean): ReadStream[WebSocketFrame] = {
+    //TODO implement this
+    rs
+  }
+
+  private def optionallyIgnorePong(rs: ReadStream[WebSocketFrame], ignore: Boolean): ReadStream[WebSocketFrame] = {
+    //TODO implement this
+      rs
+    }
+
+  private def optionallyAutoPing(rs:ReadStream[WebSocketFrame], autoPing: Option[(FiniteDuration, WebSocketFrame.Ping)]): ReadStream[WebSocketFrame] = {
+    //TODO implement this
```

Comment:
The integration works well without these implementations?

Reply:
but of course it will much better to implement those, I just haven't had time yet

Reply:
it might actually work on our node since I tested with [websockeBodyRaw](https://github.com/softwaremill/tapir/blob/dbbdd1cb4e51b4425eb35b35810451631c625020/core/src/main/scala/sttp/tapir/Tapir.scala#L203-L216), I need to test with [websocketBody](https://github.com/softwaremill/tapir/blob/dbbdd1cb4e51b4425eb35b35810451631c625020/core/src/main/scala/sttp/tapir/Tapir.scala#L167-L202) as this later is setting those values to `true`

Reply:
Edit: It does work with `websocketBody` :rocket:

Reply:
I think it's totally fine to leave it unimplemented since it's the same by other and it works.